var/spack/repos/builtin/packages/flint/package.py

```diff
@@ -0,0 +1,63 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Flint(Package):
+    """FLINT (Fast Library for Number Theory)."""
+
+    homepage = "http://www.flintlib.org"
+    url      = "http://mirrors.mit.edu/sage/spkg/upstream/flint/flint-2.5.2.tar.gz"
+
+    version('2.5.2', 'cda885309362150196aed66a5e0f0383')
+    version('2.4.5', '6504b9deabeafb9313e57153a1730b33')
+    version('develop', git='https://github.com/fredrik-johansson/flint2')
+
+    variant('mpir', default=True,
+            description='Compile with the MPIR library')
+
+    # Build dependencies
+    depends_on('autoconf', type='build')
+
+    # Other dependencies
+    depends_on('gmp',  when='~mpir')
+    depends_on('mpfr')
```

Comment:
This should also depend on `mpir` when `mpir` is used. Also `MPC`, `Piranha`, `SymEngine` and all the downstream projects. I think it would be easier, if `mpir` is not on by default.